src/wp-includes/blocks.php

```diff
@@ -1145,3 +1145,118 @@ function build_query_vars_from_query_block( $block, $page ) {
 	}
 	return $query;
 }
+
+/**
+ * Enqueue a stylesheet for a specific block.
+ *
+ * If the theme has opted-in to separate-styles loading,
+ * then the stylesheet will be enqueued on-render,
+ * otherwise when the block inits.
+ *
+ * @param string $block_name The block-name, including namespace.
+ * @param array  $args       An array of arguments [handle,src,deps,ver,media].
+ *
+ * @return void
+ */
+function wp_enqueue_block_style( $block_name, $args ) {
+	$args = wp_parse_args(
+		$args,
+		array(
+			'handle' => '',
+			'src'    => '',
+			'deps'   => array(),
+			'ver'    => false,
+			'media'  => 'all',
+		)
+	);
+
+	/**
+	 * Callback function to register and enqueue styles.
+	 *
+	 * @param string $content When the callback is used for the render_block filter,
+	 *                        the content needs to be returned so the function parameter
+	 *                        is to ensure the content exists.
+	 *
+	 * @return string
+	 */
+	$callback = function( $content ) use ( $args ) {
```

Comment:
Anonymous functions are not allowed in Core as filter/action callbacks as they are not able to be unhooked. Use a named function instead. See Core handbook https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#closures-anonymous-functions.

Reply:
@aristath: Once this PR is committed, could you please open up a new PR to address this bit of feedback?

Reply:
The WP Hooks system is not a perfect system, it has many, many flaws. I don't see how a named function could be used in this case... that's why it's using a closure.
The reason we don't allow anonymous functions as the callback in filters is that they can't be unhooked and therefore users can't "undo" the function. But in this case, the anonymous function just enqueues a stylesheet. True, the function itself can't get unhooked - but what the function does can be undone just as easily since users can use `wp_dequeue_style`.

> Once this PR is committed, could you please open up a new PR to address this bit of feedback?

No... I can't. If I could, I would have done it in the initial implementation. The anonymous function wasn't used "just because", the reason it was used is that a named function couldn't be used - or at least I don't see how, without converting this function to a class (which would be overkill for something like this).
If someone else knows how to do that then I'll watch and learn as they do it 👍

Reply:
@aristath Excellent points. Can a comment be added to share why this is needed? 

Also please use `static function` for improved performance.

Reply:
Added these in https://github.com/WordPress/wordpress-develop/pull/1804/files#r744573165 and https://github.com/WordPress/wordpress-develop/pull/1804/files#r744576658 👍 

@noisysocks can you please apply the suggested changes when you have a minute? For some reason I was unable to submit a PR against this branch today, so I added them as suggestions instead.